integration/mediation-tests/tests-platform/tests-rabbitmq/src/test/java/org/wso2/carbon/esb/rabbitmq/transport/recovery/test/RabbitMQReceiverConnectionRecoveryTestCase.java

```diff
@@ -15,115 +15,163 @@
         Thread.sleep(30000);
 
         //Stop rabbitmq server
-        rabbitMQServer.stop();
+        RabbitMQTestUtils.stopRabbitMq();
 
         //Recovery time is 10000(retry interval) * 5 (retry count) ms. Therefore continue within recovery time.
         Thread.sleep(30000);
 
         //Restart the server
-        rabbitMQServer.start();
-        rabbitMQServer.initialize();
+        RabbitMQTestUtils.startRabbitMq();
+        Thread.sleep(10000);
 
         //publish another 10 messages to broker and wait for ESB to pick up the messages
+        sender = new RabbitMQProducerClient("localhost", 5672, "guest", "guest");
         publishMessages(10);
         Thread.sleep(30000);
+        logReader.stop();
 
-        LogEvent[] logs = logViewer.getAllRemoteSystemLogs();
-        int afterLogSize = logs.length;
-        int count = 0;
-
-        for (int i = (afterLogSize - beforeLogSize - 1); i >= 0; i--) {
-            String message = logs[i].getMessage();
-            if (message.contains("received = true")) {
-                count++;
-            }
-        }
+        Assert.assertEquals(logReader.getNumberOfOccurencesForLog("received = true"),
+                            20, "All messages are not received from queue");
 
         //All 20 messages (10 messages before restarting and 10 messages after restarting)
-        Assert.assertEquals(count, 20, "Connection recovery has an error");
-
-        consumer.declareAndConnect("exchange2", "queue2");
+        consumer = new RabbitMQConsumerClient("localhost");
+        consumer.declareAndConnect("exchange2", SIMPLE_CONSUMER_QUEUE);
         Assert.assertEquals(consumer.popAllMessages().size(), 0, "All messages are not received from ESB");
+    }
+
+    @Test(groups = {
+            "wso2.esb"}, description = "Test ESB as a RabbitMQ Consumer with connection recovery when the RabbitMQ "
+                                       + "server is down when the proxy service is deployed and it doesn't come up "
+                                       + "before the recovery interval")
+    public void testRabbitMQFailureConnectionRetry() throws Exception {
+        logReader.start();
 
-        rabbitMQServer.stop();
+        //Stop rabbitmq server
+        RabbitMQTestUtils.stopRabbitMq();
+
+        //Re-deploy the proxy service
+        undeployProxyService(PROXY_NAME);
+        deployProxyService(PROXY_NAME, SOURCE_DIR);
+
+        //Recovery time is 10000(retry interval) * 5 (retry count) = 50000 ms. Therefore the RabbitMQ server starts
+        // after the recovery time.
+        Thread.sleep(70000);
+
+        //Start rabbitmq server
+        RabbitMQTestUtils.startRabbitMq();
+        Thread.sleep(10000);
+        logReader.stop();
+        String retryLog = "Attempting to create connection to RabbitMQ Broker in 10000 ms";
+
+        //Assert connection delay and retry attempts
+        Assert.assertTrue(logReader.checkForLog(retryLog, 5), "The connection retry delay is incorrect");
```

Comment:
we don't need a timeout 5 here since we stopped the logs already

Reply:
logReader.stop() only stops the tailer. It does not guarantee that that the log might be already updated just after the method call. Therefore I prefer to leave it. Besides, there will not be an impact if the log is already updated. WDYT?

Reply:
+1